nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -764,7 +796,15 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
                                             }
                                             currentValue = dest;
                                         } else if (currentValue instanceof String) {
-                                            currentValue = ((String) currentValue).getBytes(StandardCharsets.UTF_8);
+                                            final String stringValue = (String) currentValue;
+
+                                            if (BINARY_STRING_FORMAT_BASE64.getValue().equals(binaryStringFormat)) {
+                                                currentValue = BaseEncoding.base64().decode(stringValue);
+                                            } else if (BINARY_STRING_FORMAT_HEX_STRING.getValue().equals(binaryStringFormat)) {
+                                                currentValue = BaseEncoding.base16().decode(stringValue.toUpperCase());
```

Comment:
The standard Java [HexFormat](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/HexFormat.html) class should be used instead of the Google class.

Reply:
Java 8 support is not a requirement for the main branch, and HexFormat was introduced in Java 17, so this should be the direction for future integration. If the goal is to support this capability for the 1.x branch, then the feature could be backported and the Google library could be used. Going forward, however, we should make use of available Java capabilities as opposed to placing more dependencies on Google Guava or other libraries.

Reply:
Yes, I'm thinking about having this change on the 1.x branch.

So not sure what is your suggestion. If we want it on 1.x branch is it okay to use the Google library?

Or you suggest it's worth the additional work to use vanilla Java on main and instead of simply doing a cherry-pick, the committer should do the code rewrite during backport to 1.x?